Source/WebCore/rendering/RenderLayerCompositor.cpp

```diff
@@ -1310,17 +1331,29 @@ void RenderLayerCompositor::computeCompositingRequirements(RenderLayer* ancestor
         willBeComposited = true;
     };
 
+
+    // Unless we leave the containing block chain, or have an animated transform,
+    // then we can continue to use the inherited backing store attachment.
+    bool allowsBackingStoreDetachingForFixed = false;
+    if ((!ancestorLayer || layer.ancestorLayerIsInContainingBlockChain(*ancestorLayer)) && !layerExtent.hasTransformAnimation)
```

Comment:
This adds an ancestor walk which isn't great, and most layers don't have fixed descendants. I think we should try to avoid this.

Reply:
It would be better to walk the ancestor chain when we encounter a fixed layer.